starlette/websockets.py

```diff
@@ -76,22 +65,19 @@ async def send(self, message: Message) -> None:
         """
         if self.application_state == WebSocketState.CONNECTING:
             message_type = message["type"]
-            if message_type not in {"websocket.accept", "websocket.close"}:
-                if not (
-                    self._have_response_extension()
-                    and message_type == "websocket.http.response.start"
-                ):
-                    raise RuntimeError(
-                        'Expected ASGI message "websocket.accept"'
-                        ' or "websocket.close", '
-                        f"but got {message_type!r}"
-                    )
+            if message_type not in {
+                "websocket.accept",
+                "websocket.close",
+                "websocket.http.response.start",
```

Comment:
I think we should be verifying that the messages that are sent correspond to protocol and enabled extensions.  You don't have to `await response()` to invoke the `send()` method.

Reply:
I made the change already,
This code is doing sanity checks on the messages it is receiving based on the state machine.
Previous version would not accept "websocket.http.response.start" as a valid message at all if the extension was not supported.
After the change, we just send it to the server and leave the sanity checking to the server.
I'm not sure, though.  maybe we should validate messages in this place.